torch_geometric/loader/neighbor_loader.py

```diff
@@ -74,14 +90,96 @@ def __init__(
             assert input_type is not None
             self.input_type = input_type
 
+        # If we are working with a `Tuple[FeatureStore, GraphStore]` object,
+        # obtain edges from GraphStore and convert them to CSC if necessary,
+        # storing the resulting representations:
+        elif isinstance(data, tuple):
+            # TODO support `FeatureStore` with no edge types (e.g. `Data`)
+            feature_store, graph_store = data
+
+            # TODO support `collect` on `FeatureStore`
+            self.node_time_dict = None
+            if time_attr is not None:
+                raise ValueError(
+                    f"'time_attr' attribute not yet supported for "
+                    f"'{data[0].__class__.__name__}' object")
+
+            # Obtain all node and edge metadata:
+            node_attrs = feature_store.get_all_tensor_attrs()
+            edge_attrs = graph_store.get_all_edge_attrs()
+
+            self.node_types = [
+                node_attr.group_name for node_attr in node_attrs
+            ]
+            self.edge_types = [edge_attr.edge_type for edge_attr in edge_attrs]
+
+            # Set other required parameters:
+            if isinstance(num_neighbors, (list, tuple)):
+                num_neighbors = {key: num_neighbors for key in self.edge_types}
+            assert isinstance(num_neighbors, dict)
+            self.num_neighbors = {
+                edge_type_to_str(key): value
+                for key, value in num_neighbors.items()
+            }
+            self.num_hops = max([len(v) for v in self.num_neighbors.values()])
+
+            assert input_type is not None
+            self.input_type = input_type
+
+            # Obtain CSC representation of graph for in-memory sampling:
+            # TODO this code will be replaced with a `GraphStore.sample` call
+            # when sampling routines are factored out to work with pyg-lib and
+            # GraphStore
+            edge_type_to_layouts: Dict[Any,
+                                       List[EdgeLayout]] = defaultdict(list)
+            for attr in edge_attrs:
+                edge_type_to_layouts[attr.edge_type].append(attr.layout)
+
+            self.colptr_dict, self.row_dict, self.perm_dict = {}, {}, {}
```

Comment:
I think this part is too complex. For now, let us just assume that `EdgeLayout.CSC` is present. In a later PR, we can add automatic conversion of formats inside `GraphStore`.

Reply:
Throwing an exception for now to clearly indicate that only CSC and CSR are supported, since I think the added flexibility of working with CSR isn't too complicated (and we can potentially extend this to work with COO in the short term as well). But I agree that in the long term, this will be deprecated in favor of `GraphStore.sample`.